### ISSUE TITLE
make: *** No rule to make target `binaryen_js'.  Stop.

### ISSUE THREAD

**Issue by AnnieXia** [WebAssembly/binaryen#5166]
I download emscripten source code, path is "../emscripten/"
binaryen source code ,root path:

https://github.com/WebAssembly/binaryen#building

run
 `git submodule init`
   `git submodule update`
`cmake . && make`
`../emscripten/emmake cmake .`

that's OK.

But,
run `../emscripten/emmake make binaryen_js`

error :
`make: *** No rule to make target `binaryen_js'.  Stop.`

need help :(

**Comment by dcodeIO** [comment]
The recipe that works for me typically is, in Binaryen's root directory:
```
mkdir embuild
cd embuild
emcmake cmake ..
emmake make -j12 binaryen_js
```
Then builds to `embuild/bin/binarayen_js.js`.

**Comment by kripken** [comment]
Yes, the issue might be `emmake cmake .` - that should be `emcmake cmake .` (`emmake => emcmake`)

See also `scripts/emcc-tests.sh` which is how the emscripten build and tests run on CI.

**Comment by AnnieXia** [comment]
> emcmake cmake ..

I have used `emcmake cmake ..` like yours
error is still :
![image](https://user-images.githubusercontent.com/19279519/197089526-e9521f7c-d450-4397-8650-edb53242f556.png)

**Comment by AnnieXia** [comment]
> 是的，问题可能是`emmake cmake .`- 应该是`emcmake cmake .`( `emmake => emcmake`)
> 
> 另请参阅`scripts/emcc-tests.sh`emscripten 构建和测试如何在 CI 上运行。

Actrually, I used `emcmake cmake .` following 
![image](https://user-images.githubusercontent.com/19279519/197090412-0a77fe82-24b9-4320-abfa-9030271700ed.png)

Sorry, my writing was wrong.

the error still exists :(

**Comment by kripken** [comment]
I think it's using an old `CMakeCache.txt` file. For example, `BUILD_LLVM_DWARF` should not be set for the emscripten port.

Try erasing that file. Also, a fresh checkout of binaryen might work.

**Comment by tlively** [comment]
Closing the issue since the problem has likely been solved. Please reopen or file a new issue if that is not the case.
